src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -199,15 +199,31 @@ public static bool VerifyPackagesOnDisk(RestoreRequest request)
         /// <summary>
         /// Calculates the hash value, used for the no-op optimization, for the request
         /// This methods handles the deduping of tools
+        /// Handles the ignoring of RestoreSettings
         /// </summary>
         public static string GetHash(RestoreRequest request)
         {
-            if (request.Project.RestoreMetadata.ProjectStyle == ProjectStyle.DotnetCliTool)
+            if (request.Project.RestoreMetadata.ProjectStyle == ProjectStyle.DotnetCliTool || request.Project.RestoreMetadata.ProjectStyle == ProjectStyle.PackageReference)
             {
+
                 var uniqueName = request.DependencyGraphSpec.Restore.First();
                 var dgSpec = request.DependencyGraphSpec.WithProjectClosure(uniqueName);
-                dgSpec.GetProjectSpec(uniqueName).RestoreMetadata.ProjectPath = null;
-                dgSpec.GetProjectSpec(uniqueName).FilePath = null;
+                var projectSpec = dgSpec.GetProjectSpec(uniqueName);
+
+                // The project path where the tool is declared does not affect restore and is only used for logging and transparency.
+                if (request.Project.RestoreMetadata.ProjectStyle == ProjectStyle.DotnetCliTool)
+                {
+                    projectSpec.RestoreMetadata.ProjectPath = null;
+                    projectSpec.FilePath = null;
+                }
+
+                //Ignore the restore settings for net core projects.
+                //This is set by default in VS while it's not set in commandline.
+                //This causes a discrepancy and the project does not cross-client no - op.MSBuild / NuGet.exe vs VS.
+                else if (request.Project.RestoreMetadata.ProjectStyle == ProjectStyle.PackageReference)
```

Comment:
yup, changed the comment a bit. 
It's needs to be applied to both legacy and net core, but it was some nomination related code that's the root of the discrepancy.